src/main/java/io/r2dbc/postgresql/client/ReactorNettyClient.java

```diff
@@ -360,7 +360,9 @@ private void handleParameterStatus(ParameterStatus message) {
         Assert.requireNonNull(socketAddress, "socketAddress must not be null");
         Assert.requireNonNull(settings, "settings must not be null");
 
-        TcpClient tcpClient = TcpClient.create(settings.getConnectionProvider()).remoteAddress(() -> socketAddress);
+        TcpClient tcpClient = TcpClient.create(settings.getConnectionProvider())
```

Comment:
Let's apply customization only if the `SocketAddress` is a `InetSocketAddress`. We don't need round-robin for unix domain sockets.